### PR TITLE
[#112] segfault on too many metrics

### PR DESCRIPTION
**Pull request by resyfer**

### CODE REVIEW

File: src/libpgexporter/yaml_configuration.c [hunk 1]
```diff
@@ -1068,23 +1068,28 @@ free_yaml_columns(yaml_column_t** columns, size_t n_columns)
 }
 
 static int
-semantics_yaml(prometheus_t* prometheus, yaml_config_t* yaml_config)
+semantics_yaml(prometheus_t* prometheus, int prometheus_idx, yaml_config_t* yaml_config)
 {
-
    for (int i = 0; i < yaml_config->n_metrics; i++)
    {
+      prometheus_t* prom = &prometheus[prometheus_idx + i];
```
Comment (jesperpedersen) [inline src/libpgexporter/yaml_configuration.c:1075]: Start with `NULL`, and assign after the check below

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by resyfer** [comment]
PTAL

File: src/libpgexporter/yaml_configuration.c [hunk 2]
```diff
@@ -1068,23 +1068,30 @@ free_yaml_columns(yaml_column_t** columns, size_t n_columns)
...
 
 static int
-semantics_yaml(prometheus_t* prometheus, yaml_config_t* yaml_config)
+semantics_yaml(prometheus_t* prometheus, int prometheus_idx, yaml_config_t* yaml_config)
 {
+   prometheus_t* prom = NULL;
 
    for (int i = 0; i < yaml_config->n_metrics; i++)
    {
+      prom = &prometheus[prometheus_idx + i];
```
Comment (jesperpedersen) [inline src/libpgexporter/yaml_configuration.c:1077]: Same issue - move it past the check

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/libpgexporter/yaml_configuration.c (hunk 2 above)
Comment (resyfer) [reply src/libpgexporter/yaml_configuration.c:1077]: oh okay, my bad, I got it.

**Comment by jesperpedersen** [comment]
Merged, and back-ported.

Thanks for your contribution !

**Comment by resyfer** [comment]
Thanks for the review!

**Pull request merged**